client/app/lib/onboarding.coffee

```diff
@@ -0,0 +1,111 @@
+# Local class Step
+class Step
+    # Retrieves properties from config Step plain object
+    constructor: (step) ->
+        if step
+            ['name', 'route', 'view'].forEach (property) =>
+                @[property] = step[property]
+
+
+    # Record handlers for 'validated' internal pseudo-event
+    onValidated: (callback) ->
+        throw new Error 'Callback parameter should be a function' \
+            unless typeof callback is 'function'
+        @validatedHandlers = @validatedHandlers or []
+        @validatedHandlers.push callback
+
+
+    # Trigger 'validated' pseudo-event
+    triggerValidated: () ->
+        if @validatedHandlers
+            @validatedHandlers.forEach (handler) =>
+                handler(@)
+
+
+    # Submit the step
+    # This method should be overriden by step given as parameter to add
+    # for example a validation step.
+    # Maybe it should return a Promise or a call a callback couple
+    # in the near future
+    submit: () ->
+        @triggerValidated()
+
+
+# Main class
+# Onboarding is the component in charge of managing steps
+module.exports = class Onboarding
+
+
+    constructor: (user, steps) ->
+        @initialize user, steps
+
+
+    initialize: (user, steps) ->
+        throw new Error 'Missing mandatory `steps` parameter' unless steps
+
+        @user = user
+        @steps = steps.map (step) =>
+            stepModel = new Step step
+            stepModel.onValidated (step) =>
+                @handleStepSubmitted step
+            return stepModel
+
+
+    # Records handler for 'stepChanged' pseudo-event, triggered when
+    # the internal current step in onboarding has changed.
+    onStepChanged: (callback) ->
+        throw new Error 'Callback parameter should be a function' \
+            unless typeof callback is 'function'
+        @stepChangedHandlers = @stepChangedHandlers or []
```

Comment:
I personally don't see the interest here while it's used only once in each class here.